File: python/protein-translation/protein_translation.py
mapping = {
    "AUG": "Methionine",
    "UUU": "Phenylalanine",
    "UUC": "Phenylalanine",
    "UUA": "Leucine",
    "UUG": "Leucine",
    "UCU": "Serine",
    "UCA": "Serine",
    "UCC": "Serine",
    "UCG": "Serine",
    "UAU": "Tyrosine",
    "UAC": "Tyrosine",
    "UGU": "Cysteine",
    "UGC": "Cysteine",
    "UGG": "Tryptophan",
    "UAA": "STOP",
    "UAG": "STOP",
    "UGA": "STOP",
}

def proteins(strand):
    start = 0
    resp = []
    while 1:
        if start + 2 >= len(strand): break
        codon = strand[start:start + 3]
        protein = mapping[codon]
        if protein == 'STOP':
            break
        else:
            resp.append(protein)
        start += 3
    return resp

File: ruby/protein-translation/protein_translation.rb
class Translation
  def of_rna(strand)
  strand.each_slice(3).map { |protein|
    puts protein
    case protein.to_sym
    when :AUG;                   :Ethionine
    when :UUU, :UUC;             :Phenylalanine
    when :UUA, :UUG;             :Leucine
    when :UCU, :UCC, :UCA, :UCG; :Serine
    when :UAU, :UAC;             :Tyrosine
    when :UGU, :UGC;             :Cysteine
    when :UGG;                   :Tryptophan
    when :UAA, :UAG, :UGA;       break
    end
  }.map(&:to_s)
end
Translation.of_rna("UUAUUG")
